### PR TITLE
Add DefaultFunc for string field

### PR DESCRIPTION
**Pull request by haoxins**
https://github.com/facebook/ent/issues/935

### CODE REVIEW

File: schema/field/field.go [hunk 1]
```diff
@@ -208,6 +208,12 @@ func (b *stringBuilder) Default(s string) *stringBuilder {
 	return b
 }
 
+// DefaultFunc sets the default value generating function of the field.
+func (b *stringBuilder) DefaultFunc(fn func() string) *stringBuilder {
+	b.desc.Default = fn()
+	return b
+}
```
Comment (a8m) [inline schema/field/field.go:215]: The function needs to be invoked by the creator and not by the schema.
```suggestion
func (b *stringBuilder) DefaultFunc(fn func() string) *stringBuilder {
	b.desc.Default = fn
	return b
}
```

Start with this change and I can help you to add it to the codegen templates.

Comment (haoxins) [reply schema/field/field.go]: done

Comment (haoxins) [reply schema/field/field.go]: cc @a8m

**Review by a8m — APPROVED** [review]
Thanks for adding this @haoxins 

Some stuff still miss in the templates, but I can add them.  I'll merge this later, and ping you to update the docs.

**Comment by nolotz** [comment]
Hey @haoxins,

in #1086 I've made the missing changes needed in the code generation to get default functions work.

This ticket looks already abandoned, so I've included your changes as well. Please let me know if you would rather commit to these changes.
Otherwise, we should consider closing this ticket and merge my request as it offers a more general solution to the problem and will work with all Descriptors.

cc @a8m 

best regards

**Comment by haoxins** [comment]
cool

**Pull request closed without merging**